internal/primitives/runtime/digest.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+// Digest item that is able to encode/decode 'system' digest items and
+// provide opaque access to other items.
+type DigestItemTypes interface {
+	PreRuntime | Consensus | Seal | Other | RuntimeEnvironmentUpdated
+}
+
+// Digest item that is able to encode/decode 'system' digest items and
+// provide opaque access to other items.
+type DigestItem any
```

Comment:
I know that we have these types already in gossamer, is the plan to just remove the old ones?

Reply:
I will probably remove the other if we don't need it, or at the very least convert to this type when integrating with db.Backend.

Reply:
Similar to my comment on the crypto types, it's my opinion that having multiple impls of the same type is not ideal and will make things harder to maintain. So either this type should be built in a way that is compatible with the only type (to the best that we can) and/or have part of the integration plan be to remove the old type

Reply:
I agree.. I just don't know if I want to change every integration point that uses `common.Hash` when I tackle #3902.  Instead of just isolating it the use cases where `db.Backend` will be used. But knowing how you feel, I agree with you.  I'll see how bad it is when doing the integration.  My goal will be to eliminate the repeated primitives.

Reply:
Yeah I think what you say is a good approach. If it is not to bad and the use cases here are isolated and separate, then I think it could potentially be fine to have both. I just was worried if there would be a lot of crossover (I'm remembering when we had multiple mock libs lol) it could get messy, but I think your way of thinking on this is a good one